packages/gatsby/src/commands/develop-process.ts

```diff
@@ -715,6 +718,26 @@ module.exports = async (program: IDevelopArgs): Promise<void> => {
       webpackActivity = null
     }
 
+    if (isSuccessful) {
+      const state = store.getState()
+      const mapOfPagesToStaticQueryHashes = mapPagesToStaticQueryHashes(
+        state,
+        stats
+      )
+
+      const publicDir = path.join(program.directory, `public`)
+
+      mapOfPagesToStaticQueryHashes.forEach(
+        async (staticQueryHashes, pagePath) => {
+          const page = state.pages.get(pagePath)
+
+          await pageDataUtil.writePageData({ publicDir }, page, {
+            staticQueryHashes,
+          })
+        }
+      )
+    }
```

Comment:
This is (right now) only place where we write/update `page-data` files in develop mode.

There should be more triggers to rewrite those files:
 - when pure data change and doesn't cause webpack recompilation - we will write query result to `.cache/json/X.json`, but we weill not update corresponsing `page-data.json` file (until there is some potentially unrelated webpack recompilation).

It feels like we need to mark `page-data.json` files as potentially stale when we run queries, and then figure out if webpack needs recompilation or not to determine when to rewrite those

Reply:
This is now resolved